help/develop/javascript-development/packages.md

```diff
@@ -0,0 +1,263 @@
+# Packages
+
+You can extend Datagrok with _packages_ that you build using the [JavaScript API]. A Datagrok package is a JavaScript or 
+TypeScript application that introduces new functionality to Datagrok or extends the existing functionality.
+
+For example, you package might create [custom viewers using D3], demonstrate [scripting functionality for Cheminformatics],
+and do even more. Check out the [public repository] for more examples of Datagrok packages.
+
+## Table of contents
+
+* [Introduction](#package-elements)
+* [Creating a package](#creating-a-package)
+* [Package structure](#package-structure)
+* [Structuring your package](#structuring-your-package)
+
+## Package elements
+
+A package typically contains the following elements:
+
+* [Functions], [viewers], [widgets], and [applications]
+* [Scripts] that can be written in R, Python, Octave, Grok, Julia, JavaScript, Node.js, or Java
+* [Database queries] and [connections]
+* [Tables]
+
+## Creating a package
+
+To develop a package on the Datagrok platform, install the [latest LTS version of Node.js] and [npm]. We strongly
+recommend that you install Node.js using [NVM] to avoid permission issues when running NPM packages globally. Check out
+the [NPM instructions] for more details.
+
+Follow these steps to create a package template:
+
+1. Install `datagrok-tools` globally to manage your Datagrok packages:
+
+```shell
+npm install datagrok-tools -g
+```
+
+2. Configure your environment:
+
+```shell
+grok config
+```
+
+You will be prompted to enter the developer keys and set the default server. Your credentials will be stored locally
+in `config.yaml`. Check out the [Datagrok configuration section] for more details.
+
+3. Go to the folder where you want to create your package and run:
+
+```shell
+grok create <package-name>
+```
+
+> **Note**: If you want to create a TypeScript package, pass the `--ts` option to the command.
+
+After you complete the work on your Datagrok package, upload it to Datagrok by running the following command:
+
+```shell
+grok publish
+```
+
+Note that by running this command, only you will see the package in Datagrok. For more information about the available
+commands, refer to the [Grok CLI section].
```

Comment:
I think we should have the command descriptions in our wiki as well, but perhaps a link to GitHub is okay, too.